spring-security-cas-extension/src/main/java/com/kakawait/spring/security/cas/client/CasStatefulInterceptor.java

```diff
@@ -0,0 +1,41 @@
+package com.kakawait.spring.security.cas.client;
+
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.HttpRequest;
+import org.springframework.http.client.ClientHttpRequestExecution;
+import org.springframework.http.client.ClientHttpRequestInterceptor;
+import org.springframework.http.client.ClientHttpResponse;
+
+import java.io.IOException;
+import java.security.Principal;
+
+/**
+ * @author Jonathan Coueraud
+ */
+@Qualifier("casClientHttpRequestInterceptor")
+public class CasStatefulInterceptor implements ClientHttpRequestInterceptor {

```

Comment:
I will be more`StatefulCasInterceptor` rather than `CasStatefulInterceptor` to keep logic from most specific to less specific, I mean `Stateful` vs `Stateless` then is about `Cas` and then is about `Interceptor`